codalab_service.py

```diff
@@ -724,76 +534,76 @@ def run_service_cmd(self, cmd, root=False, service='rest-server'):
         if root:
             uid = '0:0'
         else:
-            uid = self.compose_env['CODALAB_UID']
+            uid = self.args.uid
         self._run_compose_cmd(
-            ('run --no-deps --rm --entrypoint="" --user=%s ' % uid)
+            ('run --no-deps --rm --user=%s ' % uid)
             + service
             + (
                 ' bash -c \'%s\'' % cmd
             )  # TODO: replace with shlex.quote(cmd) once we're on Python 3
         )
 
     def start_services(self):
-        if self.args.external_db_url is None:
+        def wait(host, port, cmd):
+            return '/opt/wait-for-it.sh {}:{} -- {}'.format(host, port, cmd)
+
+        def wait_mysql(cmd):
+            return wait(self.args.mysql_host, self.args.mysql_port, cmd)
+
+        def wait_rest_server(cmd):
+            return wait('rest-server', self.args.rest_port, cmd)
+
+        mysql_url = 'mysql://{}:{}@{}:{}/{}'.format(
+            self.args.mysql_username,
+            self.args.mysql_password,
+            self.args.mysql_host,
+            self.args.mysql_port,
+            self.args.mysql_database,
+        )
+        rest_url = 'http://rest-server:{}'.format(self.args.rest_port)
+
+        if self.args.mysql_host == 'mysql':
             self.bring_up_service('mysql')
-            cmd_prefix = '/opt/wait-for-it.sh mysql:3306 -- '
-            mysql_url = 'mysql://%s:%s@mysql:3306/codalab_bundles' % (
-                self.compose_env['CODALAB_MYSQL_USER'],
-                self.compose_env['CODALAB_MYSQL_PWD'],
-            )
-        else:
-            cmd_prefix = ''
-            mysql_url = 'mysql://%s:%s@%s/codalab_bundles' % (
-                self.compose_env['CODALAB_MYSQL_USER'],
-                self.compose_env['CODALAB_MYSQL_PWD'],
-                self.args.external_db_url,
-            )
 
         if should_run_service(self.args, 'init'):
-            print_header('Configuring the service')
-            self.run_service_cmd(
-                "%scl config server/engine_url %s && cl config cli/default_address http://rest-server:2900 && cl config server/rest_host 0.0.0.0"
-                % (cmd_prefix, mysql_url),
-                root=(not self.args.codalab_home),
-            )
+            print_header('Populating config.json')
+            commands = []
+            for config_prop, value in [
```

Comment:
This code might look more straightforward if you used a list comprehension to assign the value of `commands`.

Reply:
Done.